src/plugins/explore/public/application/utils/state_management/utils/redux_persistence.ts

```diff
@@ -270,7 +271,7 @@ const getPreloadedQueryEditorState = async (
     },
     promptModeIsAvailable,
     promptToQueryIsLoading: false,
-    editorMode: DEFAULT_EDITOR_MODE,
+    editorMode: promptModeIsAvailable ? EditorMode.Prompt : DEFAULT_EDITOR_MODE,
```

Comment:
doesn't the redux state get wiped out on each dataset change. unfortunately i think it's the current problem with the implementation pattern.

there's not singular place where this logic is occurring. we do have middleware. like is it expected to default AI for every dataset change?

Reply:
https://github.com/user-attachments/assets/9ef6c2d7-9b78-4be6-a151-253811f98300

This is the current behavior. The middleware doesn't reset the editor mode on dataset changes. So if change dataset and if user set to PPL, then it won't default to AI mode.